packages/twenty-server/src/database/commands/upgrade-version/0-31/0-31-add-index-key-to-tasks-and-notes-views.command.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  ActiveWorkspacesCommandOptions,
+  ActiveWorkspacesCommandRunner,
+} from 'src/database/commands/active-workspaces.command';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { ObjectMetadataEntity } from 'src/engine/metadata-modules/object-metadata/object-metadata.entity';
+import { TwentyORMGlobalManager } from 'src/engine/twenty-orm/twenty-orm-global.manager';
+import { ViewWorkspaceEntity } from 'src/modules/view/standard-objects/view.workspace-entity';
+
+@Command({
+  name: 'upgrade-0.31:add-index-key-to-tasks-and-notes-views',
+  description: 'Add index key to tasks and notes views',
+})
+export class AddIndexKeyToTasksAndNotesViewsCommand extends ActiveWorkspacesCommandRunner {
+  constructor(
+    @InjectRepository(Workspace, 'core')
+    protected readonly workspaceRepository: Repository<Workspace>,
+    @InjectRepository(ObjectMetadataEntity, 'metadata')
+    private readonly objectMetadataRepository: Repository<ObjectMetadataEntity>,
+    private readonly twentyORMGlobalManager: TwentyORMGlobalManager,
+  ) {
+    super(workspaceRepository);
+  }
+
+  async executeActiveWorkspacesCommand(
+    _passedParam: string[],
+    _options: ActiveWorkspacesCommandOptions,
+    workspaceIds: string[],
+  ): Promise<void> {
+    this.logger.log('Running command to fix migration');
+
+    for (const workspaceId of workspaceIds) {
+      this.logger.log(`Running command for workspace ${workspaceId}`);
+
+      try {
+        this.logger.log(chalk.green(`Cleaning views of ${workspaceId}.`));
+
+        await this.addIndexKeyToTasksAndNotesViews(
+          workspaceId,
+          _options.dryRun ?? false,
+        );
+
+        await this.twentyORMGlobalManager.destroyDataSourceForWorkspace(
+          workspaceId,
+        );
+      } catch (error) {
+        this.logger.log(
+          chalk.red(
+            `Running command on workspace ${workspaceId} failed with error: ${error}`,
+          ),
+        );
+        continue;
+      } finally {
+        this.logger.log(
+          chalk.green(`Finished running command for workspace ${workspaceId}.`),
+        );
+      }
+
+      this.logger.log(chalk.green(`Command completed!`));
+    }
+  }
+
+  private async addIndexKeyToTasksAndNotesViews(
+    workspaceId: string,
+    dryRun: boolean,
+  ): Promise<void> {
+    const viewRepository =
+      await this.twentyORMGlobalManager.getRepositoryForWorkspace<ViewWorkspaceEntity>(
+        workspaceId,
+        'view',
+        false,
+      );
+
+    const allViews = await viewRepository.find();
+
+    const viewObjectMetadataIds = allViews.map((view) => view.objectMetadataId);
+
+    const objectMetadataEntities = await this.objectMetadataRepository.find({
+      where: {
+        id: In(viewObjectMetadataIds),
+      },
+    });
+
+    const tasksAndNotesObjectMetadataIds = objectMetadataEntities.filter(
+      (entity) =>
+        entity.namePlural === 'tasks' || entity.namePlural === 'notes',
```

Comment:
nit: usually better to use standardId to identify standard objects. For a command though it should be fine since they are not expected to change name in the meantime